packages/flutter_tools/lib/src/device.dart

```diff
@@ -175,11 +175,27 @@ class DeviceManager {
   Future<List<Device>> findTargetDevices(FlutterProject flutterProject) async {
     List<Device> devices = await getDevices().toList();
 
-    if (devices.length > 1 && !deviceManager.hasSpecifiedAllDevices && !deviceManager.hasSpecifiedDeviceId) {
-      devices = devices
-          .where((Device device) => isDeviceSupportedForProject(device, flutterProject))
-          .toList();
+    // Always remove web and fuchsia devices from `--all`.
+    if (hasSpecifiedAllDevices) {
+      devices = <Device>[
+        for (Device device in devices)
+          if (await device.targetPlatform != TargetPlatform.fuchsia
+          && await device.targetPlatform != TargetPlatform.web_javascript)
```

Comment:
Nit: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#align-expressions